library/Icingadb/Web/Controller.php

```diff
@@ -566,6 +567,111 @@ public function export(Query ...$queries)
             return true;
         }
 
+        if ($this->format === 'json' || $this->format === 'csv') {
+            $isJsonFormat = $this->format === 'json';
+            $response = $this->getResponse();
+            $fileName = $this->view->title;
+
+            if ($isJsonFormat) {
+                $response
+                    ->setHeader('Content-Type', 'application/json')
+                    ->setHeader('Cache-Control', 'no-store')
+                    ->setHeader(
+                        'Content-Disposition',
+                        'attachment; filename=' . $fileName . '.json'
+                    )
+                    ->sendResponse();
+            } else {
+                $response
+                    ->setHeader('Content-Type', 'text/csv')
+                    ->setHeader('Cache-Control', 'no-store')
+                    ->setHeader(
+                        'Content-Disposition',
+                        'attachment; filename=' . $fileName . '.csv'
+                    )
+                    ->sendResponse();
+            }
+
+            ob_end_clean();
+            Environment::raiseExecutionTime();
+            Environment::raiseMemoryLimit();
+
+            $query = $queries[0];
+            $query->limit(Url::fromRequest()->getParam('limit'));
+            $col = array_merge((array) $query->getModel()->getKeyName(), $query->getModel()->getColumns());
+            $tableName = $query->getModel()->getTableName();
+
+            foreach ($query->getWith() as $relationPath => $relation) {
+                $relatedCols = $relation->getTarget()->getColumns();
+                foreach ($relatedCols as $alias => $name) {
+                    if (is_int($alias)) {
+                        $alias = $name;
+                    }
+
+                    $col[] = $relationPath . '.' . $alias;
+                }
+            }
+
+            if ($isJsonFormat) {
+                echo('[');
+            }
+
+            $rs = $query->execute()->disableCache();
+
+            foreach ($rs as $i => $row) {
+                $result = [];
+                if ($i > 0) {
+                    $separator = $isJsonFormat ? ',' : "\r\n";
+                    echo($separator);
+                }
+
+                foreach ($col as $alias => $name) {
+                    if (is_int($alias)) {
+                        $alias = $name;
+                    }
+                    if (0 === strpos($alias, $tableName)) {
```

Comment:
You are checking here whether the table name of the query models (history) is at position `0`. But might there not be such an alias? `host.state.*`